signature/mechanism_sequoia.go

```diff
@@ -0,0 +1,130 @@
+//go:build containers_image_sequoia
+
+package signature
+
+import (
+	"github.com/containers/image/v5/signature/sequoia"
+	"github.com/containers/storage/pkg/homedir"
+	"os"
+	"path"
+)
+
+// A GPG/OpenPGP signing mechanism, implemented using Sequoia.
+type sequoiaSigningMechanism struct {
+	inner *sequoia.SigningMechanism
+}
+
+// newGPGSigningMechanismInDirectory returns a new GPG/OpenPGP signing mechanism, using optionalDir if not empty.
+// The caller must call .Close() on the returned SigningMechanism.
+func newGPGSigningMechanismInDirectory(optionalDir string) (signingMechanismWithPassphrase, error) {
+	// For compatibility reasons, we allow both sequoiaHome and
+	// gpgHome to be the same directory as designated by
+	// optionalDir or GNUPGHOME.
+	envHome := os.Getenv("GNUPGHOME")
+
+	gpgHome := optionalDir
+	if gpgHome == "" {
+		gpgHome = envHome
+		if gpgHome == "" {
+			gpgHome = path.Join(homedir.Get(), ".gnupg")
+		}
+	}
+
+	sequoiaHome := optionalDir
+	if sequoiaHome == "" {
+		sequoiaHome = envHome
+		if sequoiaHome == "" {
+			dataHome, err := homedir.GetDataHome()
+			if err != nil {
+				return nil, err
+			}
+			sequoiaHome = path.Join(dataHome, "sequoia")
+		}
+	}
+
+	mech, err := sequoia.NewMechanismFromDirectory(sequoiaHome)
+	if err != nil {
+		return nil, err
+	}
+
+	// Migrate GnuPG keyrings if exist.
+	for _, keyring := range []string{"pubring.gpg", "secring.gpg"} {
+		blob, err := os.ReadFile(path.Join(gpgHome, keyring))
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return nil, err
+			}
+		} else if _, err := mech.ImportKeys(blob); err != nil {
+			return nil, err
```

Comment:
Probably wrap with something that indicates this is a failure to import a specific file.